src/main/java/no/nav/pensjon/selvbetjeningopptjening/opptjening/Pid.java

```diff
@@ -4,355 +4,64 @@
 import java.time.format.DateTimeFormatter;
 import java.time.format.DateTimeParseException;
 
-import org.apache.commons.lang3.StringUtils;
-
+import static java.lang.String.format;
+import static no.nav.pensjon.selvbetjeningopptjening.person.PidValidator.getDatePart;
+import static no.nav.pensjon.selvbetjeningopptjening.person.PidValidator.isValidPid;
+
+/**
+ * Represents a personal identification number (PID).
+ * The PID types supported are: Fødselsnummer (FNR), D-nummer (DNR) and BOST-nummer (BNR).
+ * BOST-nummer is an historical NAV-internal PID type.
+ */
 public class Pid {
-    private String pid;
 
-    public Pid(String pid) throws PidValidationException {
-        this.pid = pid;
-        validate(false);
+    private static final String DATO_PATTERN = "ddMMyyyy";
+    private final String value;
+    private LocalDate foedselsdato;
+
+    public Pid(String value) throws PidValidationException {
+        this(value, false);
     }
 
-    public Pid(String pid, boolean acceptSpecialCircumstances) throws PidValidationException {
-        this.pid = pid;
+    public Pid(String value, boolean acceptSpecialCircumstances) throws PidValidationException {
+        this.value = value;
         validate(acceptSpecialCircumstances);
     }
 
     public String getPid() {
-        return pid;
+        return value;
     }
 
+    //TODO This is not a reliable way of getting fødselsdato,
+    // since the date part of the PID is not always the fødselsdato
+    @Deprecated
     public LocalDate getFodselsdato() {
-        // Adjust bnr or dnr (for fnr return value will be equal to pid)
-        String adjustedFnr = makeDnrOrBostnrAdjustments(pid);
+        if (foedselsdato != null) {
+            return foedselsdato;
+        }
 
-        String dateString = adjustedFnr.substring(0, 4) + get4DigitYearOfBirthWithAdjustedFnr(adjustedFnr, this.isDnummer());
         try {
-            return LocalDate.parse(dateString, DateTimeFormatter.ofPattern("ddMMyyyy"));
+            return foedselsdato = LocalDate.parse(getDatePart(value), DateTimeFormatter.ofPattern(DATO_PATTERN));
```

Comment:
Tryner denne hvis man får returnert `-1` fra feks year?

Reply:
Mulig, men jeg greier ikke å lage en test for det.